<launch>
  <arg name="INPUT" default="/tilt_scan" />
  <group ns="tilt_multi_plane_detection">
    <node name="tilt_scan_to_cloud"
          pkg="laser_filters" type="scan_to_cloud_filter_chain">
      <remap from="scan" to="$(arg INPUT)" />
      <rosparam>
        scan_filter_chain:
        - name: shadows
          type: ScanShadowsFilter
          params:
            min_angle: 0
            max_angle: 170
            neighbors: 5
            window: 1
        - name: dark_shadows
          type: LaserScanIntensityFilter
          params: 
            lower_threshold: 100
            upper_threshold: 10000
            disp_histogram: 0
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet" name="tilt_laser_manager"
          args="manager" output="screen" />
    <node name="euclidean_clustering"
          pkg="nodelet" type="nodelet"
          args="load jsk_pcl/EuclideanClustering tilt_laser_manager"
          output="screen">
      <remap from="~input" to="cloud_filtered" />
    </node>
    <node pkg="nodelet" type="nodelet"
          name="euclidean_clustering_decomposer"
          args="load jsk_pcl/ClusterPointIndicesDecomposerZAxis tilt_laser_manager"
          output="screen">
      <remap from="~input" to="cloud_filtered" />
      <remap from="~target" to="euclidean_clustering/output" />
      <rosparam>
        publish_clouds: false
        publish_tf: false
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet"
          name="line_segment_detector"
          args="load jsk_pcl/LineSegmentDetector tilt_laser_manager"
          output="screen">
      <remap from="~input" to="cloud_filtered" />
      <remap from="~input_indices" to="euclidean_clustering/output" />
    </node>
    <node pkg="nodelet" type="nodelet"
          name="line_segment_decomposer"
          args="load jsk_pcl/ClusterPointIndicesDecomposerZAxis tilt_laser_manager"
          output="screen">
      <remap from="~input" to="cloud_filtered" />
      <remap from="~target" to="line_segment_detector/output/indices" />
      <rosparam>
        publish_clouds: false
        publish_tf: false
      </rosparam>
    </node>
  </group>
</launch>
